.github/workflows/deploy-docs.yml

```diff
@@ -5,8 +5,6 @@ on:
   release:
     type: [published]
   push:
-    branches:
-      - master 
```

Comment:
Does this remove pushing the updates to the master branch? Or does this do something else?

Reply:
That's exactly what it does!

Reply:
Is there any reason you are removing this? Does pushing to the master branch trigger this workflow?

Reply:
I'm removing this because we only need to trigger the workflow when there are changes to the source code (which is what the API documentation is based on) or when there are new doc pages (in the `doc` folder). We don't need to trigger this workflow when we add/edit workflows or configuration files.